### PR TITLE
Devimint use random ports

### PR DESCRIPTION
**Pull request by Maan2003**
current status: everything works!

### CODE REVIEW

File: devimint/src/external.rs [hunk 1]
```diff
@@ -33,6 +34,26 @@ pub struct Bitcoind {
 impl Bitcoind {
     pub async fn new(processmgr: &ProcessManager) -> Result<Self> {
         let btc_dir = utf8(&processmgr.globals.FM_BTC_DIR);
+        let conf = format!(
+            "\
+regtest=1
```
Comment (dpc) [inline devimint/src/external.rs:39]: `r#" ... "#';` raw strings might work better, possibly move to another file and `include_str!(...);` it anyway.

Comment (Maan2003) [reply devimint/src/external.rs]: surprising that include_str! works with format!

**Comment by dpc** [comment]
Is green? Do we land?

File: devimint/src/vars.rs [hunk 2]
```diff
@@ -86,8 +91,24 @@ declare_vars! {
         FM_TMP_DIR: PathBuf = mkdir(test_dir.into()).await?;
         FM_TEST_DIR: PathBuf = FM_TMP_DIR.clone();
         FM_TEST_FAST_WEAK_CRYPTO: String = "1";
-
         FM_LOGS_DIR: PathBuf = mkdir(FM_TEST_DIR.join("logs")).await?;
+
+        FM_PORT_BTC_RPC: u16 = port_alloc(1)?;
```
[the remark this answers is not in the document]
Comment (Maan2003) [reply devimint/src/vars.rs:96]: 60s should be more than enough. devimint starts in <15s for me locally, and all port bindings happens even before that. 4x margin is good enough.

**Comment by Maan2003** [comment]
now green, and also running tests without isolation.
I want to test multiple times to make sure that I didn't add any flakiness.

**Review by dpc — APPROVED** [review]
:tada: :tada: :tada:

**Comment by tvolk131** [comment]
Does this fix #3161?

**Comment by fedimint-backports** [comment]
Backport failed for `releases/v0.1`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally.
```bash
git fetch origin releases/v0.1
git worktree add -d .worktree/backport-3247-to-releases/v0.1 origin/releases/v0.1
cd .worktree/backport-3247-to-releases/v0.1
git checkout -b backport-3247-to-releases/v0.1
ancref=$(git merge-base 086f4ba989c7914ccc3c207ff0bd35a03695252c 95895845a5d0b7da2806cc896e84f93f97876cf9)
git cherry-pick -x $ancref..95895845a5d0b7da2806cc896e84f93f97876cf9
```

**Pull request merged**